include/boost/integer/mod_inverse.hpp

```diff
@@ -22,26 +20,26 @@ namespace boost { namespace integer {
 // Would mod_inverse be sometimes mistaken as the modular *additive* inverse?
 // In any case, I think this is the best name we can get for this function without agonizing.
 template<class Z>
-boost::optional<Z> mod_inverse(Z a, Z modulus)
+Z mod_inverse(Z a, Z modulus)
 {
-    if (modulus < 2)
+    if (modulus < Z(2))
     {
         BOOST_THROW_EXCEPTION(std::domain_error("mod_inverse: modulus must be > 1"));
     }
     // make sure a < modulus:
     a = a % modulus;
-    if (a == 0)
+    if (a == Z(0))
     {
         // a doesn't have a modular multiplicative inverse:
-        return boost::none;
+        return Z(0);
     }
-    boost::integer::euclidean_result_t<Z> u = boost::integer::extended_euclidean(a, modulus);
-    if (u.gcd > 1)
+    boost::integer::euclidean_result_t<Z> u = extended_euclidean(a, modulus);
```

Comment:
Fully qualifying is preferable since you don't intend this to be some form of ADL customization point. i.e. I would keep the `boost::integer::extended_euclidean(a, modulus)` unless you have reasons to prefer otherwise. Rest looks fine to me.

Reply:
Yup, this somehow got messed up in the merge. Thanks!